node/actors/bft/src/testonly/ut_harness.rs

```diff
@@ -0,0 +1,300 @@
+use crate::{
+    io::{InputMessage, OutputMessage},
+    leader::{ReplicaCommitError, ReplicaPrepareError},
+    replica::{LeaderCommitError, LeaderPrepareError},
+    Consensus,
+};
+use rand::{rngs::StdRng, Rng};
+use zksync_concurrency::{ctx, ctx::Ctx, scope};
+use zksync_consensus_network::io::ConsensusInputMessage;
+use zksync_consensus_roles::validator::{
+    self, BlockHeader, CommitQC, ConsensusMsg, LeaderCommit, LeaderPrepare, Payload, Phase,
+    PrepareQC, ReplicaCommit, ReplicaPrepare, SecretKey, Signed, ViewNumber,
+};
+use zksync_consensus_utils::pipe::DispatcherPipe;
+
+#[cfg(test)]
+pub(crate) struct UTHarness {
+    ctx: Ctx,
+    rng: StdRng,
```

Comment:
Don't store ctx in struct. Pass it around as an argument.
Don't store rng but rather call ctx.rng() on demand (if it is too expensive to create rng on demand for a given use case, pass it around as an argument).

Reply:
> Don't store ctx in struct. Pass it around as an argument.

I tried to minimize test code flexibility/responsibility when working with this struct, which suppose to be instantiated once per test. Is there a good reason to expose `ctx` in this case? 

> Don't store rng but rather call ctx.rng() on demand (if it is too expensive to create rng on demand for a given use case, pass it around as an argument).

I noticed the docs on using it on-demand, but I thought it’s not relevant for tests, which can also benefit determinism.

Reply:
`ctx` already has a deterministic test rng. And you are already using the test ctx, so why not use the rng provided?